### DISCUSSION TITLE
Install from and publish to a private repository

### DISCUSSION THREAD

**Discussion by bernardkaplan** [pdm-project/pdm#1473] [Q&A]
Hello,

Is it possible to install a package from a private package source and conversely to publish to a private repository (Gitlab in my personal use case) ?

**Comment by pawamoy** [comment]
Some information here:

- https://pdm.fming.dev/latest/usage/project/#publish-the-project-to-pypi
- https://pdm.fming.dev/latest/usage/dependency/#set-pypi-index-url

**Reply by baggiponte** [reply]
How do I configure the settings to upload the wheels to s3?
